app/services/email_confirmation_token_validator.rb

```diff
@@ -36,7 +36,7 @@ def confirmation_period_expired?
   end
 
   def self.email_address_from_token(token)
-    return if token.blank?
+    return if token.blank? || token.include?("\x00")
```

Comment:
https://github.com/18F/identity-idp/blob/main/app/forms/openid_connect_token_form.rb#L67
https://github.com/18F/identity-idp/blob/main/app/models/service_provider.rb#L25

For future work maybe we should add a Middleware or something to just reject bodies that contain null bytes? Instead of having to patch every entry point

Reply:
I don't like the whack-a-mole approach. The specific problem is user-submitted string that we feed to a WHERE query in postgres, and my concern is removing null bytes across the board could have unintended consequences when forms are being submitted with files or something.

Reply:
So in my mind, file uploads would be different, because we'd have to do `params[:x].read` to get the actual content. I was thinking more like something that walked the string GET and POST params and just checked those?